### PR TITLE
Shopeditor: Settings Rework

### PR DESCRIPTION
**Pull request by ZenBre4ker**
This will be a draft, that I will update from now on.
First is the ConVar:
Instead of setting random shop to 0, there is now a direct boolean to set it for.

If you find time, you might wanna check code structure after major changes. The changes might get big after a while.
Here is the first smaller one.

### CODE REVIEW

**Comment by ZenBre4ker** [comment]
Update: Random Shop settings now available in F1 Menu
grafik

**Comment by TimGoll** [comment]
Great! A few suggestions.

First help box: Explain what the random shop is and how it works.
First check box: "Enable Random Shops"
Third help box: I this the information can be merged with the checkbox label, seems redundant
Fourth help box: same as third

In general: Either add more text to the helpboxes or remove them. Right now it seems unnecessary.

**Comment by TimGoll** [comment]
Ah and don't use caps for all words. I did this too early on, but it is mostly a style used in scientific papers, not in games. So "Enable random shops" it is

**Comment by ZenBre4ker** [comment]
As Tim wants to change the way the menu is handled and I dont find motivation and time to proceed in this matter right now.
This WIP is now at least ready for merging. It contains the new ConVar and settings are already built into F1 Menu.
Next would be either item- or shop-configuration. Will probably continue after Tim reworked his part.

File: gamemodes/terrortown/gamemode/client/cl_help/populate_administration.lua [hunk 1]
```diff
@@ -77,7 +77,52 @@ ttt2net.OnUpdateGlobal({"hud_manager", "restrictedHUDs"}, function()
...
 local function PopulateRandomShopPanel(parent)
-
+	local form = vgui.CreateTTT2Form(parent, "header_random_shop_administration")
+	local form2 = vgui.CreateTTT2Form(parent, "header_random_shop_value_administration")
+
+	for convar, data in SortedPairsByMemberValue(ShopEditor.cvars, "order", false) do
+		local convarName = tostring(convar)
+		local name = "shopeditor_name_" .. data.name
+		local desc = "shopeditor_desc_" .. data.name
+		if data.typ == "bool" then
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_help/populate_administration.lua:87]: add an empty line in front

File: gamemodes/terrortown/gamemode/client/cl_help/populate_administration.lua [hunk 2]
```diff
@@ -77,7 +77,52 @@ ttt2net.OnUpdateGlobal({"hud_manager", "restrictedHUDs"}, function()
...
+			if data.b_desc then
+				form:MakeHelp({
+					label = desc
+				})
+			end
+			form:MakeCheckBox({
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_help/populate_administration.lua:93]: add an empty line in front

File: gamemodes/terrortown/gamemode/client/cl_help/populate_administration.lua [hunk 3]
```diff
@@ -77,7 +77,52 @@ ttt2net.OnUpdateGlobal({"hud_manager", "restrictedHUDs"}, function()
...
+					net.SendToServer()
+				end
+			})
+		elseif data.typ == "number" then
+			if data.b_desc then
+				form2:MakeHelp({
+					label = desc
+				})
+			end
+			form2:MakeSlider({
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/client/cl_help/populate_administration.lua:110]: add an empty line in front

File: gamemodes/terrortown/gamemode/shared/lang/de.lua [hunk 4]
```diff
@@ -1349,3 +1349,16 @@ L.karma_enemyhurt_tooltip = "Schaden"
 L.karma_cleanround_tooltip = "Saubere Runde"
 L.karma_roundheal_tooltip = "Rundenheilung"
 L.karma_unknown_tooltip = "Unbekannt"
+
+-- 2021-05-07
+L.header_random_shop_administration = "Zufalls-Shop"
+L.header_random_shop_value_administration = "Balance Einstellungen"
+
+L.shopeditor_name_random_shops = "Aktiviere Zufalls-Shop"
+L.shopeditor_desc_random_shops = "Zufalls-Shops geben jedem Spieler eine zufällige Menge an Ausrüstung.\nTeam-Shops zwingen alle Spieler aus einem Team zu ein und demselben Shopinhalt.\nAuswürfeln erlaubt es ein gegebenes Set für Credits durch ein neues Zufallsset auszutauschen."
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/shared/lang/de.lua:1358]: I think this should work like this, same holds true for the other languages
```suggestion
L.shopeditor_desc_random_shops = [[Zufalls-Shops geben jedem Spieler eine zufällige Menge an Ausrüstung.
Team-Shops zwingen alle Spieler aus einem Team zu ein und demselben Shopinhalt.
Auswürfeln erlaubt es ein gegebenes Set für Credits durch ein neues Zufallsset auszutauschen.]]
```

File: gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua [hunk 5]
```diff
@@ -14,11 +14,30 @@ ShopEditor.savingKeys = {
...
 ShopEditor.cvars = {
-	ttt2_random_shops = {typ = "number", bits = 8},
-	ttt2_random_team_shops = {typ = "bool"},
-	ttt2_random_shop_reroll = {typ = "bool"},
-	ttt2_random_shop_reroll_cost = {typ = "number", bits = 8},
-	ttt2_random_shop_reroll_per_buy = {typ = "bool"}
+	ttt2_random_shops = {
+		order = 1, typ = "bool", default = 0,
+		name = "random_shops", b_desc = true
+		},
```
Comment (TimGoll) [inline gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua:20]: I think this indentation should be one less for the brackets

**Review by TimGoll — CHANGES_REQUESTED** [review]
just some small nitpicks

File: gamemodes/terrortown/gamemode/shared/lang/de.lua (hunk 4 above)
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/lang/de.lua:1358]: Oh gotta test this. Looks way better.

File: CHANGELOG.md [hunk 6]
```diff
@@ -16,6 +16,7 @@ All notable changes to TTT2 will be documented here. Inspired by [keep a changel
   - Enables the possibility to give Innocents access to a custom shop (`shopeditor`)
 - Karma now stores changes
   - Is shown in roundend menu
+- Added Convar `ttt2_random_shop_items` for the number of items in randomshop
```
Comment (Alf21) [inline CHANGELOG.md:19]:
```suggestion
- Added the ConVar `ttt2_random_shop_items` for the number of items in the randomshop
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -31,6 +32,8 @@ All notable changes to TTT2 will be documented here. Inspired by [keep a changel
 - Code cleanup and removed silly negations
 - Extended some ttt2net functions
 - Changed `bees` win to `nones` win
+- Changed Convar `ttt2_random_shops` to only disable the random shop
```
Comment (Alf21) [inline CHANGELOG.md:35]:
```suggestion
- Changed ConVar `ttt2_random_shops` to only disable the random shop (if set to `0`)
```

File: gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua [hunk 8]
```diff
@@ -14,11 +14,30 @@ ShopEditor.savingKeys = {
...
-	ttt2_random_team_shops = {typ = "bool"},
-	ttt2_random_shop_reroll = {typ = "bool"},
-	ttt2_random_shop_reroll_cost = {typ = "number", bits = 8},
-	ttt2_random_shop_reroll_per_buy = {typ = "bool"}
+	ttt2_random_shops = {
+		order = 1, typ = "bool", default = 0,
+		name = "random_shops", b_desc = true
+	},
+	ttt2_random_shop_items = {
+		order = 2, typ = "number", bits = 8, default = 10, min = 1, max = 60,
```
Comment (Alf21) [inline gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua:22]: 2 ^ 8 = 256 items. Setting max to 60 doesn't makes sense tho. E.g. 2^6 would be enough for that limit as well. Same below.

**Review by Alf21 — CHANGES_REQUESTED** [review]

File: gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua (hunk 8 above)
Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua:22]: Well my idea was mainly, that the shop can show 5 x 6 = 30 items at once and is nice, having 60 was just double that.
Didnt know what the bits were used. Though that is a normal integer length, so I let it stay at 8 and set the max to something I though useful for a random shop.
So rather 6 bits or max 256?

Comment (ZenBre4ker) [reply gamemodes/terrortown/gamemode/shared/sh_shopeditor.lua]: Well "bits" dont matter much in my implementation. So some of those params can be deleted, once the shopeditor is fully moved to the F1 Menu. Decided to set them down to 6 then.

**Review by TimGoll — APPROVED** [review]

**Review by Alf21 — APPROVED** [review]

**Pull request merged**